components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -686,30 +370,28 @@ void AIChatTabHelper::MakeAPIRequestWithConversationHistoryUpdate(
     question_part = turn.text;
   }
 
-  std::string prompt;
-  if (UsesLlama2PromptTemplate(ai_chat::features::kAIModelName.Get())) {
-    prompt = BuildLlama2Prompt(question_part);
-  } else {
-    prompt = BuildClaudePrompt(question_part, is_suggested_question);
-  }
-
-  if (turn.visibility != ConversationTurnVisibility::HIDDEN) {
-    AddToConversationHistory(turn);
-  }
+  // Suggested questions were based on only the initial prompt (with content),
+  // so no need to submit all conversation history when they are used.
+  std::vector<mojom::ConversationTurn> history =
+      is_suggested_question ? std::vector<mojom::ConversationTurn>()
+                            : chat_history_;
 
-  DCHECK(ai_chat_api_);
   auto data_received_callback = base::BindRepeating(
-      &AIChatTabHelper::OnAPIStreamDataReceived, weak_ptr_factory_.GetWeakPtr(),
-      current_navigation_id_);
+      &AIChatTabHelper::OnEngineCompletionDataReceived,
+      weak_ptr_factory_.GetWeakPtr(), current_navigation_id_);
 
   auto data_completed_callback =
-      base::BindOnce(&AIChatTabHelper::OnAPIStreamDataComplete,
+      base::BindOnce(&AIChatTabHelper::OnEngineCompletionComplete,
                      weak_ptr_factory_.GetWeakPtr(), current_navigation_id_);
 
+  engine_->SubmitHumanInput(is_video_, article_text_, history, question_part,
```

Comment:
Perhaps. I've just kept it the same name as it is already for now, since the main purpose of this refactor is to simply extract code from the TabHelper, to independent classes. A future refactor could pass a Conversation struct to this function, and I think that next refactor will come about when we implement server-side prompt fetching.